website/views/pages.py

```diff
@@ -0,0 +1,65 @@
+from django.shortcuts import render
+from website.models import *
+from django.contrib.auth.decorators import login_required
+from .tools import get_website_section, get_latest_news_posts
+
+
+# Definition of views:
+
+def index(request):
+    context = {}
+    home_header = get_website_section('home_header')
+    getting_started = get_website_section('getting_started')
+    latest_news = get_latest_news_posts(5)
+    highlighted_publications = Publication.objects.filter(is_highlighted=True)
+    all_honeycomb_posts = list(HoneycombPost.objects.all())
+
+    context['home_header'] = home_header
+    context['getting_started'] = getting_started
+    context['latest_news'] = latest_news
+    context['highlighted_publications'] = highlighted_publications
+    context['fill_honeycomb_posts'] = []
+
+    hlength = len(all_honeycomb_posts)
+    # maximum number of honeycomb posts to display
+    max_honeycombs = 21
+    for i in range(max_honeycombs):
+        context['fill_honeycomb_posts'].append(
+            all_honeycomb_posts[i % hlength])
+
+    return render(request, 'website/index.html', context)
+
+
+def installation(request):
+    context = {}
+    installation_section = get_website_section('installation_section')
+
+    context['installation_section'] = installation_section
+
+    return render(request, 'website/installation.html', context)
+
+
+def overview(request):
```

Comment:
So what I have done now is made two types of content.

1. fixed sections - that cannot be added or deleted, only edited. and they appear at some predefined positions in the website

2. pages that can be added, deleted, edited and linked from the nav bar. This will allow us to create any page and we will get a link corresponding to that page.
Eg: if we create a page called "installation" then we get a page link http://dipy.herokuapp.com/page/installation
Also there is a checkbox to specify if we want that page in the navbar.